### DISCUSSION TITLE
Hi Rector Team. How to skip everything by default in RectorConfig and choose which files to process?

### DISCUSSION THREAD

**Discussion by alexandreelise** [rectorphp/rector#7404] [Q&A] [answered]
Hi,

I am aware that there is the skip method. But I was wondering if rector can skip everything by default and let me choose what I want to process. Is there something I am missing?

Thanks for your help @TomasVotruba

**Comment by TomasVotruba** [comment]
Hi,  how do you run Rector now?

**Reply by alexandreelise** [reply]
This is the config right now that I think might work.

```

<?php

declare(strict_types=1);

use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;
use Rector\Set\ValueObject\SetList;


return static function (RectorConfig $rectorConfig): void {
	$rectorConfig->parallel(120, 32, 20);
	
	// PHP minimum for Joomla! 4.x
	$rectorConfig->phpVersion(70205);
	
	$rectorConfig->bootstrapFiles([
		__DIR__ . '/bootstrap.php',
	]);
	
	$rectorConfig->autoloadPaths([
		__DIR__ . '/rector-autoload.php',
	]);
	
	// define sets of rules
	$rectorConfig->sets([
		LevelSetList::UP_TO_PHP_72,
		SetList::DEAD_CODE,
		SetList::MYSQL_TO_MYSQLI,
		SetList::NAMING,
	]);
	
	$rectorConfig->skip(
		[
			'*',
		]
	);
	
};



```

**Reply by TomasVotruba** [reply]
Thanks. How do you Rector from command line? 
There you can specify paths to process.

The best practice is to use paths() method in `rector.php`

**Reply by alexandreelise** [reply]
I used to use paths() method but it does not seem to find my autoloaded classes. My real question is: does rector ignore everything by default? In other words, when I put something in paths is it the ONLY files that are processed by rector?

**Reply by alexandreelise** [reply]
my command line is :

```
YOUR_JOOMLA_DIRECTORY=. composer exec rector process --dry-run --debug --ansi

```

**Reply by alexandreelise** [reply]
because an environment var is necessary for the bootstrap file to work as expected

**Comment by alexandreelise** [comment, marked as answer]
```

YOUR_JOOMLA_DIRECTORY=. composer exec rector -- process --dry-run --debug --ansi

```
Typo in the command I forgot the double dash before process.

**Comment by alexandreelise** [comment]
Hi Rector Team. You can mark this question as solved.
